File: src/scaffolds/backends/springboot/src/main/java/com/example/common/exception/BusinessException.java
package com.example.common.exception;

import com.example.common.result.ResultCode;
import lombok.Getter;

/**
 * 业务异常类
 */
@Getter
public class BusinessException extends RuntimeException {
    
    /**
     * 错误码
     */
    private final Integer code;
    
    /**
     * 错误消息
     */
    private final String message;
    
    /**
     * 使用枚举构造
     */
    public BusinessException(ResultCode resultCode) {
        super(resultCode.getMessage());
        this.code = resultCode.getCode();
        this.message = resultCode.getMessage();
    }
    
    /**
     * 自定义消息
     */
    public BusinessException(Integer code, String message) {
        super(message);
        this.code = code;
        this.message = message;
    }
    
    /**
     * 自定义消息（使用默认错误码）
     */
    public BusinessException(String message) {
        super(message);
        this.code = ResultCode.ERROR.getCode();
        this.message = message;
    }
}

File: src/scaffolds/backends/dotnet/Exceptions/BusinessException.cs
using DotnetMysqlBackend.Utils;

namespace DotnetMysqlBackend.Exceptions
{
    /// <summary>
    /// 业务异常
    /// </summary>
    public class BusinessException : Exception
    {
        public int Code { get; }

        public BusinessException(ResultCode resultCode)
            : base(resultCode.GetMessage())
        {
            Code = (int)resultCode;
        }

        public BusinessException(ResultCode resultCode, string message)
            : base(message)
        {
            Code = (int)resultCode;
        }

        public BusinessException(string message)
            : base(message)
        {
            Code = (int)ResultCode.Error;
        }

        public BusinessException(int code, string message)
            : base(message)
        {
            Code = code;
        }
    }
}
